backend/molgenis-emx2-semantics/src/main/java/org/molgenis/emx2/beaconv2/endpoints/individuals/QueryIndividuals.java

```diff
@@ -0,0 +1,118 @@
+   *
+   * @param tables
+   * @param filters
+   * @return
+   */
+  public static List<IndividualsResultSets> queryIndividuals(
+      List<Table> tables, String... filters) {
+    List<IndividualsResultSets> resultSetsList = new ArrayList<>();
+
+    StringBuffer concatFilters = new StringBuffer();
+    for (String filter : filters) {
+      concatFilters.append(filter + ",");
+    }
+    concatFilters.deleteCharAt(concatFilters.length() - 1);
+
+    for (Table table : tables) {
+      List<IndividualsResultSetsItem> individualsItemList = new ArrayList<>();
+
+      GraphQL grapql = new GraphqlApiFactory().createGraphqlForSchema(table.getSchema());
+      ExecutionResult executionResult =
+          grapql.execute(
+              "{Individuals"
+                  + "(filter: { _and: [ "
+                  + concatFilters
+                  + " ] }  )"
+                  + "{"
+                  + "id,"
+                  + "sex{name,codesystem,code},"
+                  + "age__ageGroup{name,codesystem,code},"
+                  + "age__age__iso8601duration,"
+                  + "diseaseCausalGenes{name,codesystem,code},"
+                  + "ethnicity{name,codesystem,code},"
+                  + "geographicOrigin{name,codesystem,code},"
+                  + "phenotypicFeatures{"
+                  + "   featureType{name,codesystem,code},"
+                  + "   modifiers{name,codesystem,code},"
+                  + "   severity{name,codesystem,code}},"
+                  + "diseases{"
+                  + "   diseaseCode{name,codesystem,code},"
+                  + "   ageOfOnset__ageGroup{name,codesystem,code},"
+                  + "   ageOfOnset__age__iso8601duration,"
+                  + "   ageAtDiagnosis__ageGroup{name,codesystem,code},"
+                  + "   ageAtDiagnosis__age__iso8601duration,"
+                  + "   familyHistory,"
+                  + "   severity{name,codesystem,code},"
+                  + "   stage{name,codesystem,code}},"
+                  + "measures{"
+                  + "   assayCode{name,codesystem,code},"
+                  + "   date,"
+                  + "   measurementVariable,"
+                  + "   measurementValue__value,"
+                  + "   measurementValue__units{name,codesystem,code},"
+                  + "   observationMoment__age__iso8601duration"
+                  + "}}}");
+
+      Map<String, Object> result = executionResult.toSpecification();
+
+      List<Map<String, Object>> individualsListFromJSON =
+          (List<Map<String, Object>>)
+              ((HashMap<String, Object>) result.get("data")).get("Individuals");
+
+      if (individualsListFromJSON != null) {
+        for (Map map : individualsListFromJSON) {
+          IndividualsResultSetsItem individualsItem = new IndividualsResultSetsItem();
+          individualsItem.setId(TypeUtils.toString(map.get("id")));
+          individualsItem.setSex(mapToOntologyTerm((Map) map.get("sex")));
```

Comment:
Why is this cast sometimes necessary?

Reply:
GraphQL api returns Object, must cast to List<Map> or Map to retrieve stuff. Cast is now moved to underlying function.